patches/0003-tcp_proxy-Add-filter-state-proxy_read_before_connect.patch

```diff
@@ -243,28 +251,11 @@ index 8445d2f762..6ebeab02b2 100644
 +        StreamInfo::FilterState::StateType::ReadOnly,
 +        StreamInfo::FilterState::LifeSpan::Connection);
    }
- 
+
    const std::string key_;
-@@ -1353,14 +1365,25 @@ TEST_P(TcpProxyDynamicMetadataMatchIntegrationTest, DynamicMetadataMatch) {
-   initialize();
- 
-   expectEndpointToMatchRoute([](IntegrationTcpClient& tcp_client) -> std::string {
--    // Break the write into two; validate that the first is received before sending the second. This
--    // validates that a downstream filter can use this functionality, even if it can't make a
--    // decision after the first `onData()`.
--    EXPECT_TRUE(tcp_client.write("p", false));
--    tcp_client.waitForData("p");
-+    // Break the write into multiple chunks; validate that the first word is received before sending
-+    // the rest. This validates that a downstream filter can use this functionality, even if it
-+    // can't make a decision after the first `onData()`.
-+    EXPECT_TRUE(tcp_client.write("pri", false));
-+    // Writing additional data in multiple chunks to show that buffering of early data in tcp_buffer
-+    // works properly
-+    EXPECT_TRUE(tcp_client.write("mary is ", false));
-+    EXPECT_TRUE(tcp_client.write("selected ", false));
-+    EXPECT_TRUE(tcp_client.write("before ", false));
-+    // check that the 1st word is returned
-+    tcp_client.waitForData("primary");
+@@ -1400,8 +1412,13 @@ TEST_P(TcpProxyDynamicMetadataMatchIntegrationTest, DynamicMetadataMatch) {
+     EXPECT_TRUE(tcp_client.write("p", false));
+     tcp_client.waitForData("p");
```

Comment:
This change is actually concerning. I have TLS tests failing locally (both TCP and HTTP) on the UpstreamWritesFirst tests which could be related.

Reply:
can you share you the command on how to run the above test for troubleshooting locally?

Reply:
This runs all the tests in that file:
```
bazel test @envoy//test/integration:tcp_proxy_integration_test
```

Reply:
@sayboras TLS certificates used in the tests have expired on April 6, 2024, so all TLS tests are failing, also the ones in the upstream Envoy code base.

Reply:
Upstream has reissued certs for main: https://github.com/envoyproxy/envoy/pull/33389

Reply:
ah it's making sense now as I ran the upstream tests once after applying the patches. Let me sync up with upstream.